### PR TITLE
[proof-new] Have mkScope agnostic to True assumptions

### PR DESCRIPTION
**Pull request by HanielB**

### CODE REVIEW

File: src/expr/proof_node_manager.cpp [hunk 1]
```diff
@@ -22,7 +22,10 @@ using namespace CVC4::kind;
 
 namespace CVC4 {
 
-ProofNodeManager::ProofNodeManager(ProofChecker* pc) : d_checker(pc) {}
+ProofNodeManager::ProofNodeManager(ProofChecker* pc)
+    : d_checker(pc), d_true(NodeManager::currentNM()->mkConst<bool>(true))
```
Comment (ajreynol) [inline src/expr/proof_node_manager.cpp:26]:
```suggestion
    : d_checker(pc), d_true(NodeManager::currentNM()->mkConst(true))
```

Comment (ajreynol) [reply src/expr/proof_node_manager.cpp:26]: Also I usually put this in the body of the constructor.

**Review by ajreynol — APPROVED** [review]
One minor request.

File: src/expr/proof_node_manager.cpp (hunk 1 above)
Comment (HanielB) [reply src/expr/proof_node_manager.cpp:26]: Any reason to put it in the body?

Comment (ajreynol) [reply src/expr/proof_node_manager.cpp:26]: looks weird calling NodeManager::currentNM() in an initializer list.

Comment (HanielB) [reply src/expr/proof_node_manager.cpp]: I see. I've changed it.

**Pull request merged**